guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
-
-Other locations could be configured in the `manifest.js` file, refer to the [Manifest Files and Directives](#manifest-files-and-directives).
-
-#### Search Paths
-
-When a file is referenced from a manifest or a helper, Sprockets searches all of the locations specified in `manifest.js` for it. You can view the search path by inspecting
-[`Rails.application.config.assets.paths`](configuring.html#config-assets-paths) in the Rails console.
+```ruby
+# Add additional assets to the asset load path.
+Rails.application.config.assets.paths << Emoji.images_path
+```
 
-#### Using Index Files as proxies for folders
+Propshaft will make all assets from the configured paths available for serving.
+During the precompilation process, Propshaft copies these assets into the
+`public/assets` directory, ensuring they are ready for production use. This is
+different from Sprockets, which required explicit inclusion of assets in the
+precompile list if they were not part of the bundled assets.
 
-Sprockets uses files named `index` (with the relevant extensions) for a special
-purpose.
+Assets can be referenced through their logical paths using helpers like
+`asset_path`, `image_tag`, `javascript_include_tag`, and other asset helper
+tags. After running `assets:precompile` in production, these logical references
+are automatically converted into their fingerprinted paths using the
+`manifest.json` file found in `public/assets`.
 
-For example, if you have a CSS library with many modules, which is stored in
-`lib/assets/stylesheets/library_name`, the file `lib/assets/stylesheets/library_name/index.css` serves as
-the manifest for all files in this library. This file could include a list of
-all the required files in order, or a simple `require_tree` directive.
+#### CSS and ERB
 
-It is also somewhat similar to the way that a file in `public/library_name/index.html` can be reached by a request to `/library_name`. This means that you cannot directly use an index file.
+Propshaft allows you to include CSS files, and you can also use ERB templates in
+your assets. This allows you to include dynamic content in your CSS files, like
+configuration values or environment-specific variables, just like in JavaScript
+files.
 
-The library as a whole can be accessed in the `.css` files like so:
+For example, you can include ERB within a CSS file like so:
 
 ```css
-/* ...
-*= require library_name
-*/
+/* app/assets/stylesheets/application.css.erb */
+body {
+  background-color: <%= Rails.env.production? ? 'black' : 'white' %>;
+}
 ```
 
-This simplifies maintenance and keeps things clean by allowing related code to
-be grouped before inclusion elsewhere.
+This will evaluate the ERB code and inject the appropriate color based on the
+environment.
 
 ### Coding Links to Assets
 
-Sprockets does not add any new methods to access your assets - you still use the
-familiar `stylesheet_link_tag`:
+Here's your updated section with the correct information for Propshaft:
```

Comment:
Is this line needed? Confused by stating a section way..

```suggestion
Here's your updated section with the correct information for Propshaft:
```

Reply:
Oops fixed.